app/models/payment/braintree.rb

```diff
@@ -5,15 +5,20 @@ def table_name_prefix
       'payment_braintree_'
     end
 
-    def write_transaction(bt_result, page_id, member_id, existing_customer, save_customer = true)
-      BraintreeTransactionBuilder.build(bt_result, page_id, member_id, existing_customer, save_customer)
+    # Why don't we have an options hash at the end?
+    # def write_transaction(bt_result, page_id, member_id, existing_customer, options = {})
+    def write_transaction(bt_result, page_id, member_id, existing_customer, save_customer = true, store_in_vault: false)
```

Comment:
having an optional positional parameter and an optional key parameter in one call signature seems pretty janky. I agree with commenter that an options hash would be good